jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/Environment.java

```diff
@@ -17,7 +17,7 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Map;
-import java.util.TreeMap;
+import java.util.concurrent.ConcurrentSkipListMap;
```

Comment:
Now this is a change I like! :)

Reply:
I kinda agree that `Environment`'s static API should have a thread-safe implementation, but the API itself only displays non-thread-safe behaviors.

We better be careful when using the `set(Environment)` call.

Reply:
@lorban I think `Environment.set()` should be removed.

It's easily replaced by the constructor of `Named` putting itself in the `Named.__environments` map.

Also, the `Named` constructor (and likely the whole class) should be private, as we don't want to create `Named` instances that are not in the `Named.__environments` map.
It would be enough to have `Environment.ensure(name, ClassLoader)`.